### DISCUSSION TITLE
Is twig working in ReportWidget? How to enable twig?

### DISCUSSION THREAD

**Discussion by SledgehammerPL** [wintercms/winter#807] [Q&A] [answered]
I try to use twig in report Widget, but no result:

in class MembersAtRisk extends ReportWidgetBase
`  $this->vars['members_list'] = Db::select('SELECT * FROM party_structure_members_at_risks_view');`

in _widget.htm:
```
<ul>
    {% for member in members_list %}
    <li>{{ member.person_data_first_name }}</li>
    {% endfor %}
</ul>
```

But the result is:

{% for member in members_list %}
{{ member.person_data_first_name }}
{% endfor %}

So I suppose thata have to enable twig... But how???

**Comment by mjauvin** [comment, marked as answer]
In backend views, you cannot use twig. Use PHP.

You can see how that's done in any of the backend widgets source code.

**Reply by SledgehammerPL** [reply]
That's why... Ok, so two days of searching was useless :D But you save my next days :D

**Reply by LukeTowers** [reply]
If you use VSCode and install the [official extension](https://marketplace.visualstudio.com/items?itemName=wintercms.winter-cms) and have the [`.vscode`](https://github.com/wintercms/winter/tree/develop/.vscode) folder in your project then it will automatically select the correct syntax highlighter based on the current file's location.

**Comment by mjauvin** [comment]
I encourage you to read the whole documentation, it's a good read! ;)

**Reply by SledgehammerPL** [reply]
It is very hard to find something that doesn't exists ;D 
Te documentation is very good if you know what you are looking for.. But at the begining it is like reading the bible and after that searching the name of mother-in-law of Abraham...  After 20 times you are still not sure if you think she was here and you can't find her or she isn't explicity named at all... And I'm almost sure that in docs there isn't "you can't use twig in backend" - just - "you can use twig only in CMS"  It means the same, but you can't find it if you ask wrong question.

**Comment by mjauvin** [comment]
Pretty sure the documentation states that backend views use PHP... if not, PRs to the docs are always welcome.

**Reply by SledgehammerPL** [reply]
Maybe later when i achieve higher level of knowledge.

**Comment by mjauvin** [comment]
ref. https://wintercms.com/docs/backend/controllers-ajax#actions-views-routing

**Comment by SledgehammerPL** [comment]
Was there many times...
Every puzzle is easy if you know the solution :)

**Comment by mjauvin** [comment]
What I am saying in the end is: it is worth going through THE WHOLE documentation (many times if needed) to get a better general understanding of the beast. It would save you many days of struggle...

**Comment by WebVPF** [comment]
@SledgehammerPL Like this:

```php
<ul>
    <?php foreach ($members_list as $member) : ?>
        <li>
            <?= $member->person_data_first_name ?>
        </li>
    <?php endforeach; ?>
</ul>
```
